src/main/java/com/google/sps/CapstoneAuth.java

```
@@ -10,74 +10,111 @@
 
-            if (!testRoom.hasProperty("allowed_users")) {
-                testRoom.setProperty("allowed_users", Arrays.asList(uid));
-                datastoreService.put(testRoom);
+            if (!testRoom.getNames().contains("allowed_users")) {
+                testRoom = Entity.newBuilder(testRoom)
+                                 .set("allowed_users",
+                                    ListValue.of(Arrays.asList(
+                                        StringValue.of(uid))))
+                                 .build();
 
+                datastore.put(testRoom);
                 return;
             }
 
-            List<String> allowedUsers =
-                (List<String>) testRoom.getProperty("allowed_users");
-            allowedUsers.add(uid);
-            testRoom.setProperty("allowed_users", allowedUsers);
-            datastoreService.put(testRoom);
+            List<Value<String>> allowedUsers =
+                testRoom.getList("allowed_users");
+
+            ListValue addAllowedUser = ListValue.newBuilder()
+                                                .set(allowedUsers)
+                                                .addValue(uid)
+                                                .build();
+
+            testRoom = Entity.newBuilder(testRoom)
+                             .set("allowed_users", addAllowedUser)
+                             .build();
+            datastore.put(testRoom);
+
         } catch (Exception e) {
             e.printStackTrace();
         }
     }
 
+    public static String getUserId(String idToken) {
+        if (currentInstance == null) {
+            currentInstance = new CapstoneAuth();
+        }
+
+        try {
+            FirebaseToken decodedToken =
+                FirebaseAuth.getInstance().verifyIdToken(idToken);
+
+            return decodedToken.getUid();
+        } catch (FirebaseAuthException e) {
+            e.printStackTrace();
```

Is this intended to swallow the exception?